instrumentation/java-streams/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/java/inputstream/InputStreamUtils.java

```diff
@@ -50,12 +53,34 @@ public static void addAttribute(Span span, AttributeKey<String> attributeKey, St
     if (span.isRecording()) {
       span.setAttribute(attributeKey, value);
     } else {
-      TRACER
-          .spanBuilder(HypertraceSemanticAttributes.ADDITIONAL_DATA_SPAN_NAME)
-          .setParent(Context.root().with(span))
-          .setAttribute(attributeKey, value)
-          .startSpan()
-          .end();
+      SpanBuilder spanBuilder =
+          TRACER
+              .spanBuilder(HypertraceSemanticAttributes.ADDITIONAL_DATA_SPAN_NAME)
+              .setParent(Context.root().with(span))
```

Comment:
Why is this `Context.root()` instead of `Context.current()`?

Reply:
Because if the `span` has ended, then the current may not actually be the parent of this additional-data. 
It could be inside the scope of another span, but we want this additional-data with response body to be present under the `span` which we are passing down from the VirtualStore

Reply:
Which `span` has ended?

Reply:
So if we had a root span -> parent span (which has the inputstream).
Now, the parent span is ended, but the inputstream.read() is not yet called, but the inputstream is passed down, and another span is created, and then the input stream is read. Then the current span where input stream is read is not the actual span under which we want to create the additional data span. So that may the reason. We would want it like:
```
root span:
  parent span(ended):
    additional span:
  new span(calls inputstream.read()):
```
I wasn't involved when this code was written, so I'm also trying to think and justify this.